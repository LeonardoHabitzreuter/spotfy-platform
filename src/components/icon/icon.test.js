import React from 'react'
import { shallow } from 'enzyme'
import Icon from './index'

describe('Icon component', () => {
  const wrapper = shallow(<Icon className='some-class' />)

  test('should render properly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('should has the same class passed through the props', () => {
    expect(wrapper.props()).toMatchObject({
      className: 'some-class'
    })
  })

  test('should call onClick when clicked', () => {
    const callback = jest.fn()
    const wrapper = shallow(<Icon onClick={callback} />)
    wrapper.simulate('click')
    expect(callback).toHaveBeenCalled()
  })
})
