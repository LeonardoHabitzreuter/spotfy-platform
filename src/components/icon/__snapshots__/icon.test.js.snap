// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component should render properly 1`] = `
<i
  className="some-class"
  onClick={[Function]}
/>
`;
